t/unittest/UnifiedApi/Controller/DynamicReports.t

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/perl
+
+=head1 NAME
+
+Iplogs
+
+=cut
+
+=head1 DESCRIPTION
+
+unit test for Iplogs
+
+=cut
+
+use strict;
+use warnings;
+use DateTime;
+use DateTime::Format::Strptime;
+use lib '/usr/local/pf/lib';
+
+BEGIN {
+    #include test libs
+    use lib qw(/usr/local/pf/t);
+    #Module for overriding configuration paths
+    use setup_test_config;
+}
+
+use Test::More tests => 28;
+use Test::Mojo;
+use Test::NoWarnings;
+my $t = Test::Mojo->new('pf::UnifiedApi');
+
+$t->get_ok('/api/v1/dynamic_reports' => json => { })
+  ->status_is(200);
+
+$t->get_ok('/api/v1/dynamic_report/authentications' => json => { })
+  ->json_is('/item/id',"authentications")
+  ->json_is('/item/type',"builtin")
+  ->status_is(200);
+  
+$t->post_ok('/api/v1/dynamic_report/authentications/search', {'Content-Type' => 'application/json'} => '{')
+  ->status_is(400);
+
+$t->post_ok(
+    '/api/v1/dynamic_report/authentications/search' => json => {
+        query => {
+            op    => 'equals',
+            field => 'auth_log.process_name',
+            value => 'bob'
+        }
+    }
```

Comment:
What happens if someone puts in the wrong field name?

Reply:
The SQL request will fail like it does in the current admin.

Reply:
The fingerbank/crud search endpoints gives detailed errors when an invalid field is given.
To be consistent with those endpoints we should add an issue to improve the error handling for this search endpoint.

Reply:
Yes, lets push this forward since its a bit tricky to build the list of the allowed columns since we'd have to analyze the columns of the SQL abstract query to know which ones are allowed which is the reason why I left it out since I didn't find a good way to do it easily and its not a regression since its already like this right now.

Reply:
rebase

Reply:
At minimum add some error handling for when a query fails.